### ISSUE TITLE
Sponge and FE

### ISSUE THREAD

**Issue by MrFreezeex** [ForgeEssentials/ForgeEssentials#1031]
Hi,
I see your post luac about sponge and I have some idea... This things will be cool for sponge. As you say you don't want FE to be a plugins... This dosent matter if you take the same idea as mytown had for mcpc+. To make a plugin (on sponge) who is a permission bridge... With this you have FE permission in sponge and you don't have To make FE a plugins but just a additional plugin (of sponge) To increase the compatibility.

Regards,
Mr_Freezeex

**Comment by luacs1998** [comment]
As mentioned in #1243, we will adopt the mixin systen designed by Mumfrey for ASM patching, in place of the current patcher.

**Comment by luacs1998** [comment]
@olee :

Just a quick question, from when I went over to #spongedev for a cuppa and had a chat with zml regarding the Sponge permissions system.

Sponge permission inheritance will be implicit, meaning the allowing of perm.node will also have the same effects as allowing perm.node.* or perm.node._ALL_. How is FE going to handle this, if we're going to be compatible with Sponge to a certain extent?

**Comment by Malkierian** [comment]
That's highly disturbing.  Why are they even modeling that? Even Bukkit wasn't that detailed.  They just provided a rather basic API.  I don't know that there is a way to make them compatible.  I sure hope olee does.

**Comment by olee** [comment]
We would have to redesign permission usage slightly at some places.
I think the only concrete problem would be for example "fe.commands.heal" and "fe.commands.heal.others".
Either one would have to allow the first and deny the second one, to only allow healing one self (bad solution).
Or we would make it "fe.commands.heal.self" and "fe.commands.heal.others" instead.
Then there should be no problem with inheritance.

I might also better talk with one of the sponge-devs regarding this one, too.
As imho FE's current permission system is the strongest, most versatile permission system for minecraft out there, the sponge-devs might want to take a look at what's possible and what can be migrated to sponge.

**Comment by luacs1998** [comment]
Get into #spongedev on IRC (espernet) and ask for zml.

**Comment by luacs1998** [comment]
Temporarily on hold, waiting on Sponge to prepare some things.

**Comment by luacs1998** [comment]
With Sponge moving into a beta phase, we should start looking into this properly.

**Comment by luacs1998** [comment]
@olee can you write a shim for the Sponge Permissions API to interface with FE permissions?

**Comment by ElPresidentePro** [comment]
This project is abandoned?

**Comment by spacebuilder2020** [comment]
Sponge is not abandoned but it is very low priority.   When the 1.10 version is released I think it will support sponge but it will not be a sponge plugin.

**Comment by luacs1998** [comment]
@spacebuilder2020 it will be. I had a whole design doc written, but it seems that I lost it...

**Comment by spacebuilder2020** [comment]
@yuuka-miya Has the sponge compatibility been completed at this stage?
